CMakeLists.txt

```diff
@@ -54,547 +54,69 @@
 
 cmake_minimum_required(VERSION 3.1.0)
 
-if (NOT target AND NOT CMAKE_BUILD_TYPE)
-  if (APPLE)
-    set(target clang.debug)
-  else()
-    set(target gcc.debug)
-  endif()
-endif()
+include(CMakeFuncs.cmake)
 
-if (target)
-  # Parse the target
-  set(remaining ${target})
-  while (remaining)
-    # get the component up to the next dot or end
-    string(REGEX REPLACE "^\\.?([^\\.]+).*$" "\\1" cur_component ${remaining})
-    string(REGEX REPLACE "^\\.?[^\\.]+(.*$)" "\\1" remaining ${remaining})
-
-    if (${cur_component} STREQUAL gcc)
-      if (DEFINED ENV{GNU_CC})
-        set(CMAKE_C_COMPILER $ENV{GNU_CC})
-      elseif ($ENV{CXX} MATCHES .*gcc.*)
-        set(CMAKE_CXX_COMPILER $ENV{CC})
-      else()
-        find_program(CMAKE_C_COMPILER gcc)
-      endif()
-
-      if (DEFINED ENV{GNU_CXX})
-        set(CMAKE_C_COMPILER $ENV{GNU_CXX})
-      elseif ($ENV{CXX} MATCHES .*g\\+\\+.*)
-        set(CMAKE_C_COMPILER $ENV{CC})
-      else()
-        find_program(CMAKE_CXX_COMPILER g++)
-      endif()
-    endif()
-
-    if (${cur_component} STREQUAL clang)
-      if (DEFINED ENV{CLANG_CC})
-        set(CMAKE_C_COMPILER $ENV{CLANG_CC})
-      elseif ($ENV{CXX} MATCHES .*clang.*)
-        set(CMAKE_CXX_COMPILER $ENV{CC})
-      else()
-        find_program(CMAKE_C_COMPILER clang)
-      endif()
-
-      if (DEFINED ENV{CLANG_CXX})
-        set(CMAKE_C_COMPILER $ENV{CLANG_CXX})
-      elseif ($ENV{CXX} MATCHES .*clang.*)
-        set(CMAKE_C_COMPILER $ENV{CC})
-      else()
-        find_program(CMAKE_CXX_COMPILER clang++)
-      endif()
-    endif()
-
-    if (${cur_component} STREQUAL msvc)
-      # TBD
-    endif()
-
-    if (${cur_component} STREQUAL unity)
-      set(unity true)
-      set(nonunity false)
-    endif()
-
-    if (${cur_component} STREQUAL nounity)
-      set(unity false)
-      set(nonunity true)
-    endif()
-
-    if (${cur_component} STREQUAL debug)
-      set(release false)
-    endif()
-
-    if (${cur_component} STREQUAL release)
-      set(release true)
-    endif()
-
-    if (${cur_component} STREQUAL coverage)
-      set(coverage true)
-      set(debug true)
-    endif()
-
-    if (${cur_component} STREQUAL profile)
-      set(profile true)
-    endif()
-
-  endwhile()
-
-  if (release)
-    set(CMAKE_BUILD_TYPE Release)
-  else()
-    set(CMAKE_BUILD_TYPE Debug)
-  endif()
-
-  if (NOT unity)
-    set(CMAKE_BUILD_TYPE ${CMAKE_BUILD_TYPE}Classic)
-  endif()
+set(openssl_min 1.0.2)
 
-endif()
+parse_target()
+
+set(CMAKE_CURRENT_BINARY_DIR
+  "${CMAKE_CURRENT_BINARY_DIR}/build/${target}.cmake")
```

Comment:
This is not going to do what we want for out of source builds. I'd suggest always using the `build` directory that our `scons` puts things in:
```
set(CMAKE_CURRENT_BINARY_DIR
  "${CMAKE_CURRENT_SOURCE_DIR}/build/${target}.cmake")
```